src/worker/browser-env/runner/test-runner/index.ts

```diff
@@ -0,0 +1,58 @@
+import crypto from "node:crypto";
+import URI from "urijs";
+import _ from "lodash";
+import NodejsEnvTestRunner from "../../../runner/test-runner";
+import RuntimeConfig from "../../../../config/runtime-config";
+import { wrapExecutionThread } from "./execution-thread";
+
+import type { ViteWorkerCommunicator } from "../../communicator";
+import type { WorkerTestRunnerRunOpts, WorkerTestRunnerCtorOpts } from "../../../runner/test-runner/types";
+import type { WorkerRunTestResult } from "../../../hermione";
+import type { BrowserHistory } from "../../../../types";
+import type { Browser } from "../../../../browser/types";
+import type { BrowserEnvRuntimeConfig } from "../../types";
+
+export class TestRunner extends NodejsEnvTestRunner {
+    #communicator: ViteWorkerCommunicator;
+    #runUuid: string = crypto.randomUUID();
+
+    constructor(opts: WorkerTestRunnerCtorOpts) {
+        super(opts);
+
+        this.#communicator = (RuntimeConfig.getInstance() as BrowserEnvRuntimeConfig).viteWorkerCommunicator;
+    }
+
+    async run(opts: WorkerTestRunnerRunOpts): Promise<WorkerRunTestResult> {
+        return super.run({ ...opts, ExecutionThreadCls: wrapExecutionThread(this.#runUuid) });
+    }
+
+    _getPreparePageActions(browser: Browser, history: BrowserHistory): (() => Promise<void>)[] {
```

Comment:
redefine method from parent in order to be able run action before user hook/test